AlertNG init 30 Second,but org over 5000，Causing Grafana to fail to start

### 
# What went wrong?


**What happened**:

  - ngalert.go initCtx, cancelFunc := context.WithTimeout(context.Background(), 30*time.Second)
  - bug org over 5000,so Initialization time exceeds 30 seconds，error info is :
  - `logger=infra.kvstore.sql t=2023-05-25T13:52:04.294781402Z level=debug msg="kvstore value not found" orgId=27163 namespace=alertmanager key=silences
logger=alertmanager org=27461 t=2023-05-25T13:52:04.295114265Z level=debug msg="notification log snapshot file doesn't exist" err="open /var/lib/grafana/alerting/27163/notifications: no such file or directory"
logger=ngalert.multiorg.alertmanager t=2023-05-25T13:52:04.322007724Z level=error msg="failed to apply Alertmanager config for org" org=27163 id=1814 error="unable to apply configuration: context deadline exceeded"
logger=infra.kvstore.sql t=2023-05-25T13:52:04.322196466Z level=debug msg="error getting kvstore value" orgId=20541 namespace=alertmanager key=notifications err="context deadline exceeded"`

**What did you expect to happen**:

  - 


### 
# How do we reproduce it?


**Step 1**:

- Open Grafana and do X 

**Step 2**:

- Now click button Y

**Step 3**:

- Wait for the browser to crash. Error message says: "Error..."


### 
# What Grafana version are you using?


9.4.3

### 
## Optional Questions:

### Is the bug inside a Dashboard Panel?


Copy the panel's ["get-help" data](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) here

### Grafana Platform?

Kubernetes

### User's OS?

_No response_

### User's Browser?

_No response_

### Is this a Regression?

None

### Are Datasources involved?

_No response_

### Anything else to add?


_No response_

The following information is relatively complete：
`logger=infra.kvstore.sql t=2023-05-25T11:19:30.997574564Z level=debug msg="kvstore value not found" orgId=20977 namespace=alertmanager key=notifications
logger=infra.kvstore.sql t=2023-05-25T11:19:31.005704834Z level=debug msg="kvstore value not found" orgId=20977 namespace=alertmanager key=silences
logger=alertmanager org=27461 t=2023-05-25T11:19:31.005977374Z level=debug msg="notification log snapshot file doesn't exist" err="open /var/lib/grafana/alerting/20977/notifications: no such file or directory"
logger=infra.kvstore.sql t=2023-05-25T11:19:31.030708037Z level=debug msg="kvstore value not found" orgId=28732 namespace=alertmanager key=notifications
logger=infra.kvstore.sql t=2023-05-25T11:19:31.038795683Z level=debug msg="kvstore value not found" orgId=28732 namespace=alertmanager key=silences
logger=alertmanager org=27461 t=2023-05-25T11:19:31.039070946Z level=debug msg="notification log snapshot file doesn't exist" err="open /var/lib/grafana/alerting/28732/notifications: no such file or directory"
logger=infra.kvstore.sql t=2023-05-25T11:19:31.064960051Z level=debug msg="kvstore value not found" orgId=28729 namespace=alertmanager key=notifications
logger=infra.kvstore.sql t=2023-05-25T11:19:31.073127707Z level=debug msg="kvstore value not found" orgId=28729 namespace=alertmanager key=silences
logger=alertmanager org=27461 t=2023-05-25T11:19:31.073418589Z level=debug msg="notification log snapshot file doesn't exist" err="open /var/lib/grafana/alerting/28729/notifications: no such file or directory"
logger=infra.kvstore.sql t=2023-05-25T11:19:31.09280216Z level=debug msg="kvstore value not found" orgId=28728 namespace=alertmanager key=notifications
logger=infra.kvstore.sql t=2023-05-25T11:19:31.094975843Z level=debug msg="error getting kvstore value" orgId=28728 namespace=alertmanager key=silences err="context deadline exceeded"
logger=ngalert.multiorg.alertmanager t=2023-05-25T11:19:31.095008252Z level=error msg="unable to create Alertmanager for org" org=28728 error="error reading file 'silences' from database: context deadline exceeded"
logger=cli t=2023-05-25T11:19:31.095333555Z level=error msg="Critical error" reason="runtime error: invalid memory address or nil pointer dereference" stackTrace="goroutine 1 [running]:\nruntime/debug.Stack()\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x65\ngithub.com/grafana/grafana/pkg/cmd/grafana-server/commands.RunServer.func3()\n\t/drone/src/pkg/cmd/grafana-server/commands/cli.go:181 +0x9c\npanic({0x3d9ae00, 0x732bb70})\n\t/usr/local/go/src/runtime/panic.go:890 +0x263\ngithub.com/grafana/grafana/pkg/services/ngalert/notifier.(*Alertmanager).ApplyConfig(0x0, 0xc00252ee40?)\n\t/drone/src/pkg/services/ngalert/notifier/alertmanager.go:232 +0xf5\ngithub.com/grafana/grafana/pkg/services/ngalert/notifier.(*MultiOrgAlertmanager).SyncAlertmanagersForOrgs(0xc001894300, {0x4e597a8, 0xc0018c2840}, {0xc001980000, 0x1300, 0x4e5aea0?})\n\t/drone/src/pkg/services/ngalert/notifier/multiorg_alertmanager.go:208 +0x772\ngithub.com/grafana/grafana/pkg/services/ngalert/notifier.(*MultiOrgAlertmanager).LoadAndSyncAlertmanagersForOrgs(0xc001894300, {0x4e597a8, 0xc0018c2840})\n\t/drone/src/pkg/services/ngalert/notifier/multiorg_alertmanager.go:140 +0xd8\ngithub.com/grafana/grafana/pkg/services/ngalert.(*AlertNG).init(0xc00018d6c0)\n\t/drone/src/pkg/services/ngalert/ngalert.go:176 +0x466\ngithub.com/grafana/grafana/pkg/services/ngalert.ProvideService(_, {_, _}, {_, _}, {_, _}, {_, _}, {0x4e708c8, ...}, ...)\n\t/drone/src/pkg/services/ngalert/ngalert.go:101 +0x5a9\ngithub.com/grafana/grafana/pkg/server.Initialize({{0x7ffc2d745c5a, 0x18}, {0x7ffc2d745c3e, 0x12}, {0xc000ca1da0, 0x6, 0x6}}, {{0x0, 0x0}, {0x0, ...}, ...}, ...)\n\t/drone/src/pkg/server/wire_gen.go:431 +0x4a45\ngithub.com/grafana/grafana/pkg/cmd/grafana-server/commands.RunServer({{0x4e1138c, 0x5}, {0x4e12630, 0xa}, {0x4e11384, 0x4}, {0x4e12620, 0xa}, 0xc000c3ecc0})\n\t/drone/src/pkg/cmd/grafana-server/commands/cli.go:228 +0xfea\ngithub.com/grafana/grafana/pkg/cmd/grafana-server/commands.ServerCommand.func1(0xc00143fb00?)\n\t/drone/src/pkg/cmd/grafana-server/commands/cli.go:97 +0xc5\ngithub.com/urfave/cli/v2.(*Command).Run(0xc0014b4480, 0xc000c3e8c0)\n\t/opt/drone/gomodcache/github.com/urfave/cli/v2@v2.3.0/command.go:163 +0x5c8\ngithub.com/urfave/cli/v2.(*App).RunContext(0xc000993ba0, {0x4e59770?, 0xc000082018}, {0xc0000740a0, 0xa, 0xa})\n\t/opt/drone/gomodcache/github.com/urfave/cli/v2@v2.3.0/app.go:313 +0xb36\ngithub.com/urfave/cli/v2.(*App).Run(...)\n\t/opt/drone/gomodcache/github.com/urfave/cli/v2@v2.3.0/app.go:224\nmain.main()\n\t/drone/src/pkg/cmd/grafana/main.go:39 +0x27d\n"
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2283ab5]

goroutine 1 [running]:
github.com/grafana/grafana/pkg/cmd/grafana-server/commands.RunServer.func3()
	/drone/src/pkg/cmd/grafana-server/commands/cli.go:182 +0x17e
panic({0x3d9ae00, 0x732bb70})
	/usr/local/go/src/runtime/panic.go:890 +0x263
github.com/grafana/grafana/pkg/services/ngalert/notifier.(*Alertmanager).ApplyConfig(0x0, 0xc00252ee40?)
	/drone/src/pkg/services/ngalert/notifier/alertmanager.go:232 +0xf5
github.com/grafana/grafana/pkg/services/ngalert/notifier.(*MultiOrgAlertmanager).SyncAlertmanagersForOrgs(0xc001894300, {0x4e597a8, 0xc0018c2840}, {0xc001980000, 0x1300, 0x4e5aea0?})
	/drone/src/pkg/services/ngalert/notifier/multiorg_alertmanager.go:208 +0x772
github.com/grafana/grafana/pkg/services/ngalert/notifier.(*MultiOrgAlertmanager).LoadAndSyncAlertmanagersForOrgs(0xc001894300, {0x4e597a8, 0xc0018c2840})
	/drone/src/pkg/services/ngalert/notifier/multiorg_alertmanager.go:140 +0xd8
github.com/grafana/grafana/pkg/services/ngalert.(*AlertNG).init(0xc00018d6c0)
	/drone/src/pkg/services/ngalert/ngalert.go:176 +0x466
github.com/grafana/grafana/pkg/services/ngalert.ProvideService(_, {_, _}, {_, _}, {_, _}, {_, _}, {0x4e708c8, ...}, ...)
	/drone/src/pkg/services/ngalert/ngalert.go:101 +0x5a9
github.com/grafana/grafana/pkg/server.Initialize({{0x7ffc2d745c5a, 0x18}, {0x7ffc2d745c3e, 0x12}, {0xc000ca1da0, 0x6, 0x6}}, {{0x0, 0x0}, {0x0, ...}, ...}, ...)
	/drone/src/pkg/server/wire_gen.go:431 +0x4a45
github.com/grafana/grafana/pkg/cmd/grafana-server/commands.RunServer({{0x4e1138c, 0x5}, {0x4e12630, 0xa}, {0x4e11384, 0x4}, {0x4e12620, 0xa}, 0xc000c3ecc0})
	/drone/src/pkg/cmd/grafana-server/commands/cli.go:228 +0xfea
github.com/grafana/grafana/pkg/cmd/grafana-server/commands.ServerCommand.func1(0xc00143fb00?)
	/drone/src/pkg/cmd/grafana-server/commands/cli.go:97 +0xc5
github.com/urfave/cli/v2.(*Command).Run(0xc0014b4480, 0xc000c3e8c0)
	/opt/drone/gomodcache/github.com/urfave/cli/v2@v2.3.0/command.go:163 +0x5c8
github.com/urfave/cli/v2.(*App).RunContext(0xc000993ba0, {0x4e59770?, 0xc000082018}, {0xc0000740a0, 0xa, 0xa})
	/opt/drone/gomodcache/github.com/urfave/cli/v2@v2.3.0/app.go:313 +0xb36
github.com/urfave/cli/v2.(*App).Run(...)
	/opt/drone/gomodcache/github.com/urfave/cli/v2@v2.3.0/app.go:224
main.main()
	/drone/src/pkg/cmd/grafana/main.go:39 +0x27d`

--- a/conf/defaults.ini
+++ b/conf/defaults.ini
@@ -1156,6 +1156,9 @@ enabled =
 # Comma-separated list of organization IDs for which to disable unified alerting. Only supported if unified alerting is enabled.
 disabled_orgs =
 
+# Specify how long to wait for the alerting service to initialize
+initialization_timeout = 30s
+
 # Specify the frequency of polling for admin config changes.
 # The interval string is a possibly signed sequence of decimal numbers, followed by a unit suffix (ms, s, m, h, d), e.g. 30s or 1m.
 admin_config_poll_interval = 60s

--- a/conf/sample.ini
+++ b/conf/sample.ini
@@ -1076,6 +1076,9 @@
 # Comma-separated list of organization IDs for which to disable unified alerting. Only supported if unified alerting is enabled.
 ;disabled_orgs =
 
+# Specify how long to wait for the alerting service to initialize
+;initialization_timeout = 30s
+
 # Specify the frequency of polling for admin config changes.
 # The interval string is a possibly signed sequence of decimal numbers, followed by a unit suffix (ms, s, m, h, d), e.g. 30s or 1m.
 ;admin_config_poll_interval = 60s

--- a/pkg/services/ngalert/ngalert.go
+++ b/pkg/services/ngalert/ngalert.go
@@ -150,7 +150,7 @@ type AlertNG struct {
 
 func (ng *AlertNG) init() error {
 	// AlertNG should be initialized before the cancellation deadline of initCtx
-	initCtx, cancelFunc := context.WithTimeout(context.Background(), 30*time.Second)
+	initCtx, cancelFunc := context.WithTimeout(context.Background(), ng.Cfg.UnifiedAlerting.InitializationTimeout)
 	defer cancelFunc()
 
 	ng.store.Logger = ng.Log

--- a/pkg/services/ngalert/tests/util.go
+++ b/pkg/services/ngalert/tests/util.go
@@ -45,7 +45,8 @@ func SetupTestEnv(tb testing.TB, baseInterval time.Duration) (*ngalert.AlertNG, 
 
 	cfg := setting.NewCfg()
 	cfg.UnifiedAlerting = setting.UnifiedAlertingSettings{
-		BaseInterval: setting.SchedulerBaseInterval,
+		BaseInterval:          setting.SchedulerBaseInterval,
+		InitializationTimeout: 30 * time.Second,
 	}
 	// AlertNG database migrations run and the relative database tables are created only when it's enabled
 	cfg.UnifiedAlerting.Enabled = new(bool)

--- a/pkg/services/quota/quotaimpl/quota_test.go
+++ b/pkg/services/quota/quotaimpl/quota_test.go
@@ -485,7 +485,7 @@ func setupEnv(t *testing.T, sqlStore *sqlstore.SQLStore, b bus.Bus, quotaService
 	ac := acimpl.ProvideAccessControl(sqlStore.Cfg)
 	ruleStore, err := ngstore.ProvideDBStore(sqlStore.Cfg, featuremgmt.WithFeatures(), sqlStore, &foldertest.FakeService{}, &dashboards.FakeDashboardService{}, ac)
 	require.NoError(t, err)
-	cfg.UnifiedAlerting.InitializationTimeout = 30 * time.Second
+	sqlStore.Cfg.UnifiedAlerting.InitializationTimeout = 30 * time.Second
 	_, err = ngalert.ProvideService(
 		sqlStore.Cfg, featuremgmt.WithFeatures(), nil, nil, routing.NewRouteRegister(), sqlStore, ngalertfakes.NewFakeKVStore(t), nil, nil, quotaService,
 		secretsService, nil, m, &foldertest.FakeService{}, &acmock.Mock{}, &dashboards.FakeDashboardService{}, nil, b, &acmock.Mock{},

--- a/pkg/services/quota/quotaimpl/quota_test.go
+++ b/pkg/services/quota/quotaimpl/quota_test.go
@@ -485,6 +485,7 @@ func setupEnv(t *testing.T, sqlStore *sqlstore.SQLStore, b bus.Bus, quotaService
 	ac := acimpl.ProvideAccessControl(sqlStore.Cfg)
 	ruleStore, err := ngstore.ProvideDBStore(sqlStore.Cfg, featuremgmt.WithFeatures(), sqlStore, &foldertest.FakeService{}, &dashboards.FakeDashboardService{}, ac)
 	require.NoError(t, err)
+	cfg.UnifiedAlerting.InitializationTimeout = 30 * time.Second
 	_, err = ngalert.ProvideService(
 		sqlStore.Cfg, featuremgmt.WithFeatures(), nil, nil, routing.NewRouteRegister(), sqlStore, ngalertfakes.NewFakeKVStore(t), nil, nil, quotaService,
 		secretsService, nil, m, &foldertest.FakeService{}, &acmock.Mock{}, &dashboards.FakeDashboardService{}, nil, b, &acmock.Mock{},

--- a/pkg/setting/setting_unified_alerting.go
+++ b/pkg/setting/setting_unified_alerting.go
@@ -43,6 +43,7 @@ const (
 	}
 }
 `
+	alertingDefaultInitializationTimeout    = 30 * time.Second
 	evaluatorDefaultEvaluationTimeout       = 30 * time.Second
 	schedulerDefaultAdminConfigPollInterval = time.Minute
 	schedulerDefaultExecuteAlerts           = true
@@ -80,6 +81,7 @@ type UnifiedAlertingSettings struct {
 	HARedisPassword                string
 	HARedisDB                      int
 	HARedisMaxConns                int
+	InitializationTimeout          time.Duration
 	MaxAttempts                    int64
 	MinInterval                    time.Duration
 	EvaluationTimeout              time.Duration
@@ -190,6 +192,11 @@ func (cfg *Cfg) ReadUnifiedAlertingSettings(iniFile *ini.File) error {
 			return err
 		}
 		uaCfg.DisabledOrgs[orgID] = struct{}{}
+	}
+
+	uaCfg.InitializationTimeout, err = gtime.ParseDuration(valueAsString(ua, "initialization_timeout", (alertingDefaultInitializationTimeout).String()))
+	if err != nil {
+		return err
 	}
 
 	uaCfg.AdminConfigPollInterval, err = gtime.ParseDuration(valueAsString(ua, "admin_config_poll_interval", (schedulerDefaultAdminConfigPollInterval).String()))

--- a/pkg/setting/setting_unified_alerting.go
+++ b/pkg/setting/setting_unified_alerting.go
@@ -175,6 +175,8 @@ func (cfg *Cfg) readUnifiedAlertingEnabledSetting(section *ini.Section) (*bool, 
 
 // ReadUnifiedAlertingSettings reads both the `unified_alerting` and `alerting` sections of the configuration while preferring configuration the `alerting` section.
 // It first reads the `unified_alerting` section, then looks for non-defaults on the `alerting` section and prefers those.
+//
+//nolint:gocyclo
 func (cfg *Cfg) ReadUnifiedAlertingSettings(iniFile *ini.File) error {
 	var err error
 	uaCfg := UnifiedAlertingSettings{}

--- a/pkg/setting/setting_unified_alerting_test.go
+++ b/pkg/setting/setting_unified_alerting_test.go
@@ -25,6 +25,7 @@ func TestCfg_ReadUnifiedAlertingSettings(t *testing.T) {
 		require.Len(t, cfg.UnifiedAlerting.HAPeers, 0)
 		require.Equal(t, 200*time.Millisecond, cfg.UnifiedAlerting.HAGossipInterval)
 		require.Equal(t, time.Minute, cfg.UnifiedAlerting.HAPushPullInterval)
+		require.Equal(t, alertingDefaultInitializationTimeout, cfg.UnifiedAlerting.InitializationTimeout)
 	}
 
 	// With peers set, it correctly parses them.
@@ -34,10 +35,13 @@ func TestCfg_ReadUnifiedAlertingSettings(t *testing.T) {
 		require.NoError(t, err)
 		_, err = s.NewKey("ha_peers", "hostname1:9090,hostname2:9090,hostname3:9090")
 		require.NoError(t, err)
+		_, err = s.NewKey("initialization_timeout", "123s")
+		require.NoError(t, err)
 
 		require.NoError(t, cfg.ReadUnifiedAlertingSettings(cfg.Raw))
 		require.Len(t, cfg.UnifiedAlerting.HAPeers, 3)
 		require.ElementsMatch(t, []string{"hostname1:9090", "hostname2:9090", "hostname3:9090"}, cfg.UnifiedAlerting.HAPeers)
+		require.Equal(t, 123*time.Second, cfg.UnifiedAlerting.InitializationTimeout)
 	}
 
 	t.Run("should read 'scheduler_tick_interval'", func(t *testing.T) {

